core/cap-0038.md

```diff
@@ -686,15 +686,68 @@ index 75f39eb4..f590ad4b 100644
 
 ### Semantics
 
+#### Asset Ordering in LiquidityPoolConstantProductParameters
+`LiquidityPoolConstantProductParameters` describes the immutable properties of a
+constant product liquidity pool. Specifically, these parameters are the two
+assets between which the pool can trade and the fee rate charged for such a
+trade. But the two assets, referred to as `assetA` and `assetB`, must satisfy an
+ordering relation to avoid the existence of functionally duplicate pools. For
+example, a pool that can exchange between `X` and `Y` is no different from a
+pool that can exchange between `Y` and `X`. The ordering relation imposed is
+field-wise lexicographic ordering, defined as follows for arbitrary XDR values
+`u` and `v` of the same type `T`:
+
+```
+bool lessThan<T>(u, v)
+    if isUnion(T)
+        if u.discriminant == v.discriminant
+            // body means the contents corresponding to this value of the
+            // discriminant
+            return lessThan(u.body, v.body)
+        else
+            return u.discriminant < v.discriminant
+    else if isStruct(T)
+        for i in 0...numFields(T)
+            if lessThan(u.field[i], v.field[i])
+                return true
+            else if lessThan(v.field[i], u.field[i])
+                return false
+            // This field was equal so compare the next one
+        // If we reached this point, all fields were equal so clearly u is not
+        // less than v
+        return false
+    else if isContainer(T)
+        for i in 0...length(T)
+            if lessThan(u[i], v[i])
+                return true
+            else if lessThan(v[i], u[i])
+                return false
+            // This element was equal so compare the next one
+        // If we reached this point, all elements were equal so clearly u is not
+        // less than v
+        return false
+    else // T is an integer type
+        return u < v
```

Comment:
what?

fwiw, I implemented this in go as roughly:
```go
func aLessThanB(a, b int) bool {
	if a.Type < b.Type {
		return true
	}

	if a.Code < b.Code {
		return true
	}

	if a.Issuer < b.Issuer {
		return true
	}

	return false
}
```

Reply:
I'll try to clarify this.

I know you described the above as "roughly" so it might not apply to your real implementation, but there is an error in this rough implementation. This doesn't correctly handle the case where `a.Type > b.Type` (and likewise for `a.Code > b.Code`). The proceeding compared value must be equal before considering the next one.

Reply:
The definition in the CAP is going to have more complexity than that simple implementation, because I want it to continue to be accurate even if `Asset` changes.